front/src/views/user/user-settings/user-settings.view.js

```diff
@@ -116,6 +126,33 @@ export default {
         .catch(error => {
           this.emailIsAlreadyTaken = error.response.data.isEmailUsed;
         });
+    },
+    avatarUpload (file) {
+      this.fileUploadError = null;
+
+      if (file && file.size < MAX_AVATAR_SIZE_KB) {
+        const formData = new FormData();
+        formData.append('file', file);
+
+        this.$store.dispatch('userSettings/uploadUserAvatarFile', formData)
+          .then(r => {
+            if (r.status === 200) {
+              this.editedUserSettings.avatarFileName = r.data;
```

Comment:
as i know the response with status 200 is always successfully resolved, so it is not necessary checking, you can write just .then(response => this.editedUserSettings.avatarFileName = r.data) 
responses with unsuccessful status will rejected and  it could be handles in catch

Reply:
thank, removed